homeassistant/components/sensor/hddtemp.py

```diff
@@ -47,11 +47,11 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     port = config.get(CONF_PORT)
     disks = config.get(CONF_DISKS)
 
-    try:
-        hddtemp = HddTempData(host, port)
-        hddtemp.update()
-    except RuntimeError:
-        _LOGGER.error("Unable to fetch the data from %s:%s", host, port)
+    hddtemp = HddTempData(host, port)
+    hddtemp.update()
+
+    if hddtemp.data is None:
+        _LOGGER.error("Error connecting ...")
```

Comment:
We can drop this error since we're already sending an error out here: https://github.com/cgtobi/home-assistant/blob/a08183ae23a598b9b8a43af9263c02f7a126717a/homeassistant/components/sensor/hddtemp.py#L137 Just returning False here should be good. (My mistake in the example I gave in chat)

Reply:
Thanks for the constructive feedback. I'll incorporate that right away.